src/develop/develop.c

```diff
@@ -1640,48 +1639,32 @@ static gboolean _dev_auto_apply_presets(dt_develop_t *dev)
     return FALSE;
   }
 
+  //  get current workflow and image characteristics
+
   const gboolean is_scene_referred = dt_is_scene_referred();
   const gboolean is_display_referred = dt_is_display_referred();
   const gboolean is_workflow_none = !is_scene_referred && !is_display_referred;
-
-  //  Add scene-referred workflow
-  //  Note that we cannot use a preset for FilmicRGB as the default values are
-  //  dynamically computed depending on the actual exposure compensation
-  //  (see reload_default routine in filmicrgb.c)
-
   const gboolean has_matrix = dt_image_is_matrix_correction_supported(image);
 
-  const char *workflow = dt_conf_get_string_const("plugins/darkroom/workflow");
+  //  set filters
 
-  const gboolean auto_apply_filmic =
-    (is_raw || is_mono) && (strcmp(workflow, "scene-referred (filmic)") == 0);
-  const gboolean auto_apply_sigmoid =
-    (is_raw || is_mono) && (strcmp(workflow, "scene-referred (sigmoid)") == 0);
-  const gboolean auto_apply_exposure =
-    auto_apply_filmic || auto_apply_sigmoid;
-  const gboolean auto_apply_basecurve =
-    (is_raw || is_mono) && (strcmp(workflow, "display-referred (legacy)") == 0);
-  const gboolean auto_apply_cat =
-    has_matrix && is_modern_chroma;
+  int iformat = 0;
+  if(dt_image_is_raw(image))
+    iformat |= FOR_RAW;
+  else
+    iformat |= FOR_LDR;
 
-  if(auto_apply_filmic || auto_apply_sigmoid || auto_apply_cat || auto_apply_basecurve)
-  {
-    for(GList *modules = dev->iop; modules; modules = g_list_next(modules))
-    {
-      dt_iop_module_t *module = (dt_iop_module_t *)modules->data;
+  if(has_matrix)
+    iformat |= FOR_MATRIX;
 
-      if(((auto_apply_exposure && dt_iop_module_is(module->so, "exposure"))
-          || (auto_apply_filmic && dt_iop_module_is(module->so, "filmicrgb"))
-          || (auto_apply_sigmoid && dt_iop_module_is(module->so, "sigmoid"))
-          || (auto_apply_basecurve && dt_iop_module_is(module->so, "basecurve"))
-          || (auto_apply_cat && dt_iop_module_is(module->so, "channelmixerrgb")))
-         && !dt_history_check_module_exists(imgid, module->op, FALSE)
-         && !(module->flags() & IOP_FLAGS_NO_HISTORY_STACK))
-      {
-        _dev_insert_module(dev, module, imgid);
-      }
-    }
-  }
+  if(dt_image_is_hdr(image))
+    iformat |= FOR_HDR;
+
+  int excluded = 0;
+  if(dt_image_monochrome_flags(image))
```

Comment:
Maybe it should just be a test for `dt_image_is_monochrome(image)` here ...

Reply:
Sorry for the noise. It all works just perfectly. I just messed up my presets while switching between versions and didn't notice.

Reply:
Good to hear :) Thanks for checking.